### DISCUSSION TITLE
Next.js `app` dir support for SIWE

### DISCUSSION THREAD

**Discussion by alex-grover** [family/connectkit#224] [Ideas]
Would love to be able to move our app over to the new route handlers from API routes.

It looks like this would be blocked by support for the app dir in `iron-session`, which is slated for their next major release: https://github.com/vvo/iron-session/issues/586

**Comment by lochie** [comment]
If you'd like a temporary workaround you can check out @m1guelpf's example of how to implement ConnectKit's Sign In With Ethereum functionality with the new [Next.js](https://nextjs.org/) App Router.

https://github.com/m1guelpf/nextjs13-connectkit-siwe

**Reply by alex-grover** [reply]
@lochie I've adapted the above into several projects at this point, and am thinking about publishing a package similar/equivalent to `connectkit-next-siwe` specifically for the app dir.

Would you be interested in a PR for that on this repo?

**Reply by lochie** [reply]
@alex-grover a PR would be awesome for this! it would be hugely appreciated 🙏

**Comment by jxom** [comment]
Can also check out the official wagmi template: https://github.com/wagmi-dev/create-wagmi/tree/main/templates/next/connectkit

**Reply by richardryangarcia** [reply]
excellent job on this release! just what i have been waiting for. 

Does this latest release support SIWE ?

**Comment by dylanngph** [comment]
I need support when using NextAuth for Connectkit Authentication, please help
